lib/cuda/cudaSyncStream.cpp

```diff
@@ -0,0 +1,52 @@
+#include "cudaSyncStream.hpp"
+
+using kotekan::bufferContainer;
+using kotekan::Config;
+
+REGISTER_CUDA_COMMAND(cudaSyncStream);
+
+cudaSyncStream::cudaSyncStream(Config& config, const std::string& unique_name,
+                               bufferContainer& host_buffers, cudaDeviceInterface& device) :
+    cudaCommand(config, unique_name, host_buffers, device, "sync", "") {
+    set_source_cuda_streams(config.get<std::vector<int32_t>>(unique_name, "source_cuda_streams"));
+    set_command_type(gpuCommandType::BARRIER);
+}
+
+cudaSyncStream::cudaSyncStream(kotekan::Config& config, const std::string& unique_name,
+                               kotekan::bufferContainer& host_buffers, cudaDeviceInterface& device,
+                               bool called_by_subclasser) :
+    cudaCommand(config, unique_name, host_buffers, device, "sync", "") {
+    (void)called_by_subclasser;
```

Comment:
This still needs a command type set for the internal profiling.  
```suggestion
    (void)called_by_subclasser;
    command_type = gpuCommandType::BARRIER;
```

Reply:
Done!